### DISCUSSION TITLE
playlist merge every 15 minutes

### DISCUSSION THREAD

**Discussion by Marcioghiraldelli** [AzuraCast/AzuraCast#7543] [General]
Hello, it seems that I discovered a bug or the counting algorithm is working wrong.

If you set a playlist to play every 15 minutes and the playlist is 17 minutes long and you mix it to play everything when the interval starts, it does not reset the 15-minute count of this playlist when the entire mix is ​​finished.

In other words, it plays for 17 minutes, a jingle starts and then plays the entire mix again.

In my opinion, this is wrong, because from the panels I tested, even the Centova Cast, which is very old, resets the count and does not repeat when you say that you want it to play everything mixed every x minutes.

This bug is more frequent when you have other intervals together, such as every x songs and x minutes and this x minutes mixed.

**Comment by BusterNeece** [comment]
I'm a little conflicted about this one, actually.

On the one hand, I can definitely understand your logic here: if a track is 5 minutes long, and you set it to play every 15 minutes, then what you might mean is that there should be 15-minute _gaps_ between track plays, so the loop would actually be 20 minutes long.

On the other, people might be actually wanting a track to _start_ every 15 minutes, regardless of how long the track itself is.

I'm not sure that there's a way to solve this for both groups without adding an additional configuration option, something I'm always hesitant to do.

**Reply by Marcioghiraldelli** [reply]
@BusterNeece So, in fact, from what I understand, the right thing to do would be to set it every 15 minutes, it comes on, plays the 5 minutes of the track, and after 15 minutes it plays again. In this case that happened, it is very complicated to understand, generating an extra calculation so that everything works correctly. Do you understand my point of view?

**Reply by Marcioghiraldelli** [reply]
Currently, I should take the total time of the playlist and add it to the time every 15 minutes, and this is quite complicated. In my opinion, the right thing to do would be, regardless of the length of the playlist in time, it will play and after 15 minutes it will play again, this way there is no need for extra calculations.

In other words, if the list is 60 minutes long and I set it to every 15 minutes, it will play for 60 minutes and then play again 15 minutes later. In my opinion, it would be easier to understand and manage.

**Reply by BusterNeece** [reply]
@Marcioghiraldelli Your proposed option actually involves _more_ code, as we track when songs were queued up and _started_ playing, not when they finish, but it's a reasonable thing to want. Just one more thing for the pile of stuff to do with the scheduler :)
